src/shared/contexts/UserContext.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {
+  createContext,
+  SetStateAction,
+  useState,
+  Dispatch,
+  useEffect,
+} from "react";
+import { Amplify, Auth } from "aws-amplify";
+import axios from "axios";
+
+export const amplifyConfig = Amplify.configure({
+  aws_cognito_region: process.env.REACT_APP_AWS_REGION,
+  aws_user_pools_id: process.env.REACT_APP_COGNITO_USER_POOL_ID,
+  aws_user_pools_web_client_id: process.env.REACT_APP_COGNITO_CLIENT_ID,
+  aws_mandatory_sign_in: "enable",
+});
+
+type CurrentUser = {
+  email: string;
+  sub?: string;
+  firstName?: string;
+  lastName?: string;
+  postCode?: string;
+  token?: string;
```

Comment:
Shall we rename this to idToken, in case we add accessTokens or refreshTokens at a later date?

(We probably should be using refresh tokens but that's definitely another ticket for another time)